### PR TITLE
Tweak docs to include policy override option

### PR DESCRIPTION
**Pull request by robbrockbank**
## Description
Documentation for the SystemNetworkPolicy resource used as a Calico-Policy override mechanism.

### CODE REVIEW

File: hack/remove-calico-policy/README.md [hunk 1]
```diff
@@ -1,4 +1,41 @@
...
+
+This guide explains two methods for disabling Calico policy on a running cluster. These steps are intended as a
+last resort should it appear that Calico policy is not functioning properly on a production system.
+
+- [Overriding Calico Policy] describes how to add high-priority Calico System Network Policy which
+  overrides the configured Network Policy.
+- [Disabling and Removing Calico Policy] describes how to fully disable Calico policy (removing all 
+  Calico-programmed policy from each node.
+
+## Overriding Calico Policy
```
Comment (caseydavenport) [inline hack/remove-calico-policy/README.md]: Let's move this to a separate markdown file, and maybe just have a link from this doc along the lines of "These steps are intended as a last resort.  You probably want to try this first"

File: hack/remove-calico-policy/README.md [hunk 2]
```diff
@@ -1,4 +1,41 @@
...
+- These steps assume Calico is running in policy-only mode (without Calico networking)
+
+### Instructions
+
+> **Note:** The following steps assume you have permissions to create resources in the `kube-system` namespace.
+
+#### 1. Create a `SystemNetworkPolicy` resource
+
+```
+kubectl create --namespace=kube-system --filename=./system-network-policy-override.yaml
```
Comment (caseydavenport) [inline hack/remove-calico-policy/README.md]: Would be great if the user didn't need to download the file first - let's make it

```
kubectl create  -f https://raw.githubusercontent.com/....`
```

Also, let's use shorthand flags and remove the `-n` and put the namespace in the file instead.

File: hack/remove-calico-policy/README.md [hunk 3]
```diff
@@ -1,4 +1,41 @@
...
+
+- [Overriding Calico Policy] describes how to add high-priority Calico System Network Policy which
+  overrides the configured Network Policy.
+- [Disabling and Removing Calico Policy] describes how to fully disable Calico policy (removing all 
+  Calico-programmed policy from each node.
+
+## Overriding Calico Policy
+
+This section describes how to override the Calico-enabled Network Policy using higher priority system-wide
+network policy.  This system-wide policy is implemented using a `SystemNetworkPolicy` Third Party Resource.
```
Comment (gunjan5) [inline hack/remove-calico-policy/README.md]: nit: `.  This` extra space after the period

File: hack/remove-calico-policy/README.md [hunk 4]
```diff
@@ -1,4 +1,41 @@
...
+
+- Calico version v2.3 or higher
+- Kubernetes v1.6 or higher
+- These steps assume Calico is running in policy-only mode (without Calico networking)
+
+### Instructions
+
+> **Note:** The following steps assume you have permissions to create resources in the `kube-system` namespace.
+
+#### 1. Create a `SystemNetworkPolicy` resource
```
Comment (gunjan5) [inline hack/remove-calico-policy/README.md]: why is this step `1.` if there's only one step?

File: hack/remove-calico-policy/README.md (hunk 3 above)
Comment (gunjan5) [inline hack/remove-calico-policy/README.md]: also, do we need to specify the implementation details? I'm not sure

Comment (robbrockbank) [inline hack/remove-calico-policy/README.md]: grrrr - it's very difficult to break that habit. lol

Comment (robbrockbank) [inline hack/remove-calico-policy/README.md]: yeah I'm not sure....  probably not for this.

File: hack/remove-calico-policy/README.md (hunk 4 above)
Comment (robbrockbank) [inline hack/remove-calico-policy/README.md]: good point.

File: hack/remove-calico-policy/README.md (hunk 1 above)
Comment (robbrockbank) [inline hack/remove-calico-policy/README.md]: If this is a separate MD then we either have the option of:
-  creating a whole new directory specifically for this option
-  add another MD which won't be called README.md (but keep everything in the same dir)
-  add a separate README.md doc and move the existing README.md to a new MD doc (so the README is the landing page and everything is in the same dir).

For general docs flow I would go with the latter.  Agree?

File: hack/remove-calico-policy/README.md (hunk 2 above)
Comment (robbrockbank) [inline hack/remove-calico-policy/README.md]: Sure - I was trying to be consistent with the existing instructions.

So you want the other instructions updated too?

Comment (gunjan5) [inline hack/remove-calico-policy/README.md]: I personally think they all should be pulled from a URL, so it's easier when/if we move this over to docs.projectcalico.org i.e. version templating can be applied to them (if needed) so they don't get out of sync

File: hack/remove-calico-policy/README.md (hunk 1 above)
Comment (robbrockbank) [inline hack/remove-calico-policy/README.md]: Or do you want the main README.md specifically to be the big switch

File: hack/remove-calico-policy/README.md (hunk 2 above)
Comment (robbrockbank) [inline hack/remove-calico-policy/README.md]: I assume this should be doc links not GitHub user content?  (i.e. links to docs.projectcalico blah)?

File: hack/remove-calico-policy/README.md (hunk 1 above)
Comment (caseydavenport) [inline hack/remove-calico-policy/README.md]: Gr, let's make this doc the big switch.

- README.md
  - override-policy.md
  - remove-policy.md

README.md can contain the intro / context bits.

File: hack/remove-calico-policy/README.md (hunk 2 above)
Comment (caseydavenport) [inline hack/remove-calico-policy/README.md]: > So you want the other instructions updated too?

Yeah, that's probably good.  Sorry about the double standard :p

File: hack/remove-calico-policy/override-policy.md [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+- These steps assume Calico is running in policy-only mode (without Calico networking)
+
+### Instructions
+
+> **Note:** The following steps assume you have permissions to create resources in the `kube-system` namespace.
+
+Create a `SystemNetworkPolicy` resource by running the following commands:
+
+```
+curl https://github.com/projectcalico/calico/blob/master/hack/remove-calico-policy/system-network-policy-override.yaml -o system-network-policy-override.yaml
```
Comment (caseydavenport) [inline hack/remove-calico-policy/override-policy.md]: Can we turn this into one command?

File: hack/remove-calico-policy/remove-policy.md [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+
+```
+curl https://github.com/projectcalico/calico/blob/master/hack/remove-calico-policy/remove-calico-policy.sh -o remove-calico-policy.sh
+kubectl create configmap remove-calico-policy-config -n=kube-system --from-file=./remove-calico-policy.sh
+```
+
+Then, deploy the DaemonSet:
+
+```
+curl https://raw.githubusercontent.com/projectcalico/calico/master/hack/remove-calico-policy/iptables-remover-ds.yaml -o iptables-remover-ds.yaml
```
Comment (caseydavenport) [inline hack/remove-calico-policy/remove-policy.md]: Can we turn this into a single command?

File: hack/remove-calico-policy/override-policy.md (hunk 5 above)
Comment (robbrockbank) [inline hack/remove-calico-policy/override-policy.md]: Actually I believe -f will work with URLs so yes.

File: hack/remove-calico-policy/remove-policy.md (hunk 6 above)
Comment (robbrockbank) [inline hack/remove-calico-policy/remove-policy.md]: same as above.   should be able to.  The only one we can't is --from-file for creating the config map

**Pull request merged**